targets/javascript/make.js

```diff
@@ -76,11 +76,20 @@ function getRequestActions(tabbing, apiCall) {
     if (apiCall.result === "LoginResult" || apiCall.request === "RegisterPlayFabUserRequest")
         return tabbing + "request.TitleId = PlayFab.settings.titleId ? PlayFab.settings.titleId : request.TitleId; if (!request.TitleId) throw PlayFab._internalSettings.errorTitleId;\n";
     if (apiCall.auth === "EntityToken")
-        return tabbing + "if (!PlayFab._internalSettings.entityToken) throw PlayFab._internalSettings.errorEntityToken;\n";
+        return tabbing + "var authKey = \"X-EntityToken\"; var authValue = PlayFab._internalSettings.GetAuthValue(request, authKey);\n"
+            + tabbing + "if (!authValue) throw PlayFab._internalSettings.errorEntityToken;\n";
     if (apiCall.auth === "SessionTicket")
-        return tabbing + "if (!PlayFab._internalSettings.sessionTicket) throw PlayFab._internalSettings.errorLoggedIn;\n";
+        return tabbing + "var authKey = \"X-Authorization\"; var authValue = PlayFab._internalSettings.GetAuthValue(request, authKey);\n"
+            + tabbing + "if (!authValue) throw PlayFab._internalSettings.errorLoggedIn;\n";
     if (apiCall.auth === "SecretKey")
-        return tabbing + "if (!PlayFab.settings.developerSecretKey) throw PlayFab._internalSettings.errorSecretKey;\n";
+        return tabbing + "var authKey = \"X-SecretKey\"; var authValue = PlayFab._internalSettings.GetAuthValue(request, authKey);\n"
+            + tabbing + "if (!authValue) throw PlayFab._internalSettings.errorSecretKey;\n";
+//    TODO: Replace the 3 apiCall.auth if-blocks above with the generic solution below once PlayFab._internalSettings.GetAuthInfo() is ready
+//    if (["EntityToken", "SessionTicket", "SecretKey"].includes(apiCall.auth))
+//        return tabbing + "var authKey = \"X-EntityToken\"; var authInfo = PlayFab._internalSettings.GetAuthInfo(request, authKey);\n"
```

Comment:
Ah, this hard-coded authKey could be the reason why PlayFab._internalSettingsGetAuthInfo() fails for the GetPlayerProfile unit test.  That definitely shouldn't be hard-coded.  Hmm...